### PR TITLE
Add static routes of net interfaces

### PR DESCRIPTION
**Pull request by EdwardStudy**
This PR allows agent support to add static routes of net interfaces. These changes have been worked well with SoftLayer.

Added 'Routes' list in the Interface configuration:

```
type Routes []Route

type Route struct {
  Destination string
  Gateway string
  Netmask string
}
```

The bosh agent got these property and applied the `post-up route add -net {{ .Destination }} netmask {{ .Netmask }} gw {{ .Gateway }}` template to the '/etc/network/interfaces' file:
```
auto eth0
iface eth0 inet static
    address 10.11.11.143
    network 10.11.11.0
    netmask 255.255.255.0
    post-up route add -net 10.0.0.0 netmask 255.0.0.0 gw 10.11.11.1
    post-up route add -net 29.0.0.0 netmask 255.0.0.0 gw 10.11.11.1
```
After ifup-ing these interfaces, the agent finished configuring VM networks with static routes successfully.

@mattcui @maximilien @cppforlife  Please help to review. Thank you~

### CODE REVIEW

**Comment by dpb587-pivotal** [comment]
🔗 Pivotal Tracker: [Story #155716437](https://www.pivotaltracker.com/story/show/155716437)

**Comment by luan** [comment]
@EdwardStudy do you mind following the bot's request for signing the CLA? We can't merge this until that's resolved.

**Comment by EdwardStudy** [comment]
@luan OK. I just sent request mail.

**Comment by EdwardStudy** [comment]
I just received CLA response. 

Close and reopen it.

**Comment by luan** [comment]
Thanks @EdwardStudy! We'll get to reviewing this PR shortly then :)

File: platform/net/ubuntu_net_manager.go [hunk 1]
```diff
@@ -359,15 +359,17 @@ auto lo
...
 {{ end }}{{ range .StaticConfigs }}
 auto {{ .Name }}
 iface {{ .Name }} inet{{ .Version6 }} static
     address {{ .Address }}{{ if not .IsVersion6 }}
     network {{ .Network }}{{ end }}
     netmask {{ .NetmaskOrLen }}{{ if .IsDefaultForGateway }}{{ if not .IsVersion6 }}
     broadcast {{ .Broadcast }}{{ end }}
-    gateway {{ .Gateway }}{{ end }}
+    gateway {{ .Gateway }}{{ end }}{{ range .PostUpRoutes }}
+    post-up route add -net {{ .Destination }} netmask {{ .Netmask }} gw {{ .Gateway }}{{ end }}
```
Comment (mikexuu) [inline platform/net/ubuntu_net_manager.go]: According to the [docs](http://manpages.ubuntu.com/manpages/artful/man8/route.8.html), it looks like when you're adding an IPv6 address with the `route` utility, you need to specify the address family to override the IPv4 default like so:

```
# route -A inet6 add <TARGET> gw <GW>
route -A inet6 add fe80::800:27ff:fe01:0/128 gw ::1
```

Can you please add logic/tests for determining the family of the IP address and using the correct route command?

Comment (mikexuu) [inline platform/net/ubuntu_net_manager.go]: It looks like the command to add a IPv6 route is slightly different - we need to override the default address family like so:

```
# route -A inet6 add <target> gw <gw>
route -A inet6 add fe80::800:27ff:fe01:0/128 gw ::1
```

Could you please add logic/tests to determine whether a route is ipv6 to use the proper command?

Comment (EdwardStudy) [reply platform/net/ubuntu_net_manager.go]: @mikexuu Yes. Only support ipv4 currently. I will add it.

**Comment by EdwardStudy** [comment]
@mikexuu Hi, Mike. Updated commit.

**Pull request merged**